services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchMaintenance.groovy

```diff
@@ -0,0 +1,210 @@
+        northbound.setSwitchMaintenance(sw.dpId, true, false)
+
+        then: "Flows are not evacuated (rerouted) and have the same paths"
+        PathHelper.convert(northbound.getFlowPath(flow1.id)) == flow1Path
+        PathHelper.convert(northbound.getFlowPath(flow2.id)) == flow2Path
+
+        when: "Set maintenance mode again with flows evacuation flag for the same switch"
+        northbound.setSwitchMaintenance(sw.dpId, true, true)
+
+        then: "Flows are evacuated (rerouted)"
+        Wrappers.wait(WAIT_OFFSET) {
+            [flow1, flow2].each { assert northbound.getFlowStatus(it.id).status == FlowState.UP }
+        }
+
+        def flow1PathUpdated = PathHelper.convert(northbound.getFlowPath(flow1.id))
+        def flow2PathUpdated = PathHelper.convert(northbound.getFlowPath(flow2.id))
+
+        flow1PathUpdated != flow1Path
+        flow2PathUpdated != flow2Path
+
+        and: "Switch under maintenance is not involved in new flow paths"
+        !(sw in pathHelper.getInvolvedSwitches(flow1PathUpdated))
+        !(sw in pathHelper.getInvolvedSwitches(flow2PathUpdated))
+
+        and: "Delete flows and unset maintenance mode"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+        northbound.setSwitchMaintenance(sw.dpId, false, false)
+    }
+
+    def "Link discovered by a switch under maintenance is marked as maintained"() {
+        given: "An active link"
+        def isl = topology.islsForActiveSwitches.first()
+
+        and: "Bring port down on the switch to fail the link"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+            assert islUtils.getIslInfo(islUtils.reverseIsl(isl)).get().state == IslChangeType.FAILED
+        }
+
+        and: "Delete the link"
+        List<DeleteLinkResult> responses = []
+        responses << northbound.deleteLink(islUtils.getLinkParameters(isl))
+        responses << northbound.deleteLink(islUtils.getLinkParameters(islUtils.reverseIsl(isl)))
+        responses.each { assert it.deleted }
+
+        when: "Set maintenance mode for the switch"
+        northbound.setSwitchMaintenance(isl.srcSwitch.dpId, true, false)
+
+        and: "Bring port up to discover the deleted link"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "The link is discovered and marked as maintained"
+        Wrappers.wait(WAIT_OFFSET) {
+            def links = northbound.getAllLinks()
+            def islInfo = islUtils.getIslInfo(links, isl).get()
+            def reverseIslInfo = islUtils.getIslInfo(links, islUtils.reverseIsl(isl)).get()
+
+            [islInfo, reverseIslInfo].each {
+                assert it.state == IslChangeType.DISCOVERED
+                assert it.underMaintenance
+            }
+        }
+
+        and: "Unset maintenance mode"
+        northbound.setSwitchMaintenance(isl.srcSwitch.dpId, false, false)
```

Comment:
It makes sense to verify that the maintenance mode is being unset from the ISL as well. I see that you did this verification in previous test, but since this is a new ISL which was not present at the start, it may behave differently